### ISSUE TITLE
Automatically Adjust Stats/Abilities When Dependencies are Changed

### ISSUE THREAD

**Issue by Thronewolf** [foundryvtt-dcc/dcc#272]
Some things I've noticed that do not get updated when dependencies change.

- To Hit and Damage for melee (or ranged) weapons does not get updated when the STR or AGI modifiers are changed.
- Initiative modifier is not updated when the AGI modifier is updated
- When a character's level changes, a number of things are not automatically updated. These include but are not limited to:

1. Character special ability stats (EX: Chaotic Thief goes from lvl 1 to lvl 2, the Thief abilities like Pick Lock, Luck Die, etc do not get updated to reflect the level 2 table. Same for Warrior and Mighty Deed die.)
2. Fort, Ref, and Will saves
3. Action Die

This can be a little confusing at first because some things on the character sheets are automated, but some aren't. Right now it's a bit of a guessing game to figure out if something else on your character sheet needs to be updated manually. Thankfully Journal entries for the class with all the relevant tables are linked on the sheets.

**Comment by vittordrow** [comment]
I'd like to voice my opinion that this indeed what I'd expect from a good System. The tables are fixed by level/class/alignment, so it would save the trouble of having to update it manually. 

Also right now many of the rolls do not get bonuses from the related rolled attributes (EX: When rolling Sneak it only adds the filled in bonus at the skill, it doesn't add AGI to it), and attributes are something that get changed temporarily quite often in DCC. Also there's only one slot to input values on those places, so even if done manually you'd have to check every time from where those numbers come (Abil scores/Class/Birth/etc). So having an way of setting up where those numbers come from would also be neat. And having the numbers from level/class/align automatically filled would be even better.

**Comment by mooped** [comment]
The sheet is currently 'direct reading'.

What you type in to the boxes is what you get. Sometimes there are options to compute those fields for you (AC, Check Penalty), but the value in the box is what gets applied.

If you import a character from Purple Sorcerer following the instructions in the User Guide this means that everything is setup correctly. The Purple Sorcerer exports have everything rolled into the numbers provided which means they just work.

If we start adding things automatically to the numbers then two problems occur:
a) Users are no longer sure if they need to take modifiers into account when entering their stats, and the rolls they see differ from what's in the box.
b) The Purple Sorcerer importer now has to reverse engineer what portion of the numbers it sees are modifers that the system will add on elsewhere, and account for those when importing a character. This can be difficult, especially when bonuses come from the character's Birth Augur or somewhere else quite freeform.

Obviously the Purple Sorcerer importer is not the only feature, but it's an extremely useful one for running one shots and funnels. For a campaign game players should be committed enough to understand how to level their characters.

Rest assured this is something we want to address and it will be happening, but other features need to happen first to avoid breaking other features.

There should be no guessing what on the sheet needs to be updated because it's everything that's not greyed out. For now the rule is that the number you see on the sheet is what gets rolled.

**Comment by algnc** [comment]
Duplicate of #148 

**Rest assured this is an option we are working on.** Any class related level up automation would have to be part of the Core Book module, not the System, due to our agreement about character sheets with Goodman Games. The system is built to be free, by volunteers, so it doesn't contain super specific information straight out of the core book. 

We've been running games using the system for a year now so we're aware of limitations, and what will make it good. I know some people have unique challenges with newer players, but we've run hundreds of sessions among us with regular players, road crew players, and convention players and so far the system has worked great. 

We have to keep in mind several things when adding features, especially around automation: 

1. Automation is a double edged sword. We've all used Fantasy Grounds and see the huge amount of extra work for a Judge when the automation doesn't work correctly.
2. We have to support this free system, as volunteers, so any issues that arise, or things people don't understand, have to be addressed through our support channels, and we already have a lot of work to do just getting people going with the system and Foundry.
3. We have to test the system against every Foundry update, and every update we release, and automation will increase the test load.
4. This feature is a tricky one to implement because it will impact extremely important features like the Purple Sorceror importer.
